diambra/arena/env_settings.py

```diff
@@ -1,200 +1,389 @@
 from dataclasses import dataclass
-from typing import Union, List, Tuple
+from typing import Union, List, Tuple, Any, Dict
 from diambra.arena.utils.gym_utils import available_games
 import numpy as np
+import random
+from diambra.engine import model
+import time
 
 MAX_VAL = float("inf")
 MIN_VAL = float("-inf")
 MAX_FRAME_RES = 512
 MAX_STACK_VALUE = 48
 
 def check_num_in_range(key, value, bounds):
-    assert (value >= bounds[0] and value <= bounds[1]), "\"{}\" ({}) value must be in the range {}".format(key, value, bounds)
+    error_message = "ERROR: \"{}\" ({}) value must be in the range {}".format(key, value, bounds)
+    assert (value >= bounds[0] and value <= bounds[1]), error_message
+    assert (type(value)==type(bounds[0])), error_message
 
 def check_val_in_list(key, value, valid_list):
-    assert (value in valid_list), "\"{}\" ({}) admissible values are {}".format(key, value, valid_list)
-
+    error_message = "ERROR: \"{}\" ({}) admissible values are {}".format(key, value, valid_list)
+    assert (value in valid_list), error_message
+    assert (type(value)==type(valid_list[valid_list.index(value)])), error_message
 
 @dataclass
 class EnvironmentSettings:
+    env_info = None
+    games_dict = None
 
+    # Fixed at startup
     game_id: str
-
-    # System level
+    frame_shape: Tuple[int, int, int] = (0, 0, 0)
     step_ratio: int = 6
-    disable_keyboard:bool = True
+    n_players: int = 1
+    disable_keyboard: bool = True
     disable_joystick: bool = True
+    render_mode: Union[str, None] = None
     rank: int = 0
-    seed: int = -1
     env_address: str = "localhost:50051"
     grpc_timeout: int = 600
 
-    # Game level
-    player: str = "Random"
+    # Variable at reset
+    seed: Union[int, None] = None
+    difficulty: Union[int, str] = "Random"
```

Comment:
Why can this be a int or str? Seems confusing

Reply:
I see your point. For some environment settings, the idea would be to allow the user to specify their value (that in general is a number) or optionally to randomize that parameter. In the case of difficulty then it would be, for example in a game where minimum is 1 and max is 4, you can specify the exact value (1, 2, 3, 4) or you may want it to be randomized at every reset. 

Being able to randomize env parameter is very important in this field in general, and while I could have used 0 for that, I thought it could be clearer for the user when is specifying the settings dictionary to explicitly say `Random` instead of 0, as that can also be confusing as there is always the topic 0- or 1-indexed values..... What do you think?

Reply:
I think better than having this being a int or the string 'Random' would be:
- Use -1 for 'random', possibly even expose consts for that (DIFFICULTY_RANDOM = -1, DIFFICULTY_EASY = 1 - though I know games have different difficulties but might still be cleaner)
- Use 'None' for random, basically if a user doesn't specify a difficulty it's random
- Add another atttribute like randomize_difficulty (or even a generic `randomize` attribute that takes a list of fields to randomize)
- Always require setting difficulty but provide a function to return a random, valid difficulty (probably requires further refactoring given you need to first parse the game for this)

I'd say just go for what option feels right, even if you feel what we currently have is better. It's alright either way

Reply:
Yes lets go for None there as well for consistency

Reply:
This should be done.